GW2EIEvtcParser/EIData/CombatReplay/CombatReplay.cs

```diff
@@ -1,649 +1,667 @@
                     else
                     {
-                        ParametricPoint3D ptn = Rotations[rotationTablePos + 1];
-                        if (ptn.Time < i)
+                        ParametricPoint3D last = PolledRotations.Last().Time > pt.Time ? PolledRotations.Last() : pt;
+                        if (ptn.Time - last.Time > ArcDPSEnums.ArcDPSPollingRate + rate)
                         {
-                            rotationTablePos++;
-                            i -= rate;
+                            PolledRotations.Add(new ParametricPoint3D(last.X, last.Y, last.Z, i));
                         }
                         else
                         {
-                            ParametricPoint3D last = PolledRotations.Last().Time > pt.Time ? PolledRotations.Last() : pt;
-                            if (ptn.Time - last.Time > ArcDPSEnums.ArcDPSPollingRate + rate)
-                            {
-                                PolledRotations.Add(new ParametricPoint3D(last.X, last.Y, last.Z, i));
-                            }
-                            else
-                            {
-                                float ratio = (float)(i - last.Time) / (ptn.Time - last.Time);
-                                PolledRotations.Add(new ParametricPoint3D(last, ptn, ratio, i));
-                            }
-
+                            float ratio = (float)(i - last.Time) / (ptn.Time - last.Time);
+                            PolledRotations.Add(new ParametricPoint3D(last, ptn, ratio, i));
                         }
+
                     }
                 }
             }
-            PolledRotations = PolledRotations.Where(x => x.Time >= 0).ToList();
         }
+        PolledRotations.RemoveAll(x => x.Time < 0);
+    }
 
-        internal void PollingRate(long fightDuration, bool forcePositionPolling)
-        {
-            PositionPolling(ParserHelper.CombatReplayPollingRate, fightDuration, forcePositionPolling);
-            RotationPolling(ParserHelper.CombatReplayPollingRate, fightDuration);
-        }
+    internal void PollingRate(long fightDuration, bool forcePositionPolling)
+    {
+        PositionPolling(ParserHelper.CombatReplayPollingRate, fightDuration, forcePositionPolling);
+        RotationPolling(ParserHelper.CombatReplayPollingRate, fightDuration);
+    }
 
 
-        #region DEBUG EFFECTS
+    #region DEBUG EFFECTS
 
-        internal static void DebugEffects(AbstractSingleActor actor, ParsedEvtcLog log, CombatReplay replay, HashSet<long> knownEffectIDs, long start = long.MinValue, long end = long.MaxValue)
+    internal static void DebugEffects(AbstractSingleActor actor, ParsedEvtcLog log, CombatReplay replay, HashSet<long> knownEffectIDs, long start = long.MinValue, long end = long.MaxValue)
+    {
+        var effectEventsOnAgent = log.CombatData.GetEffectEventsByDst(actor.AgentItem)
+            .Where(x => !knownEffectIDs.Contains(x.EffectID) && x.Time >= start && x.Time <= end);
```

Comment:
The purpose of the unused variables in the debug methods it to have an easy access to certain values when breaking in that code, I'd very much prefer that they remain here, I use it to isolate effects and their GUIDs within certain windows.

Reply:
Added them back with a comment about this usage, and a define to toggle the functions of for production builds